sdk/python/kfp/dsl/types.py

```diff
@@ -19,65 +19,76 @@ class BaseType:
 
 # Primitive Types
 class Integer(BaseType):
-	openapi_schema_validator = {
-		"type": "integer"
-	}
+	def __init__(self):
```

Comment:
I kind of liked the previous variant better: A "singleton" `Integer` type that you do not need to instantiate.

`def my_component(a : Integer, b: GCSPath)`

Reply:
The previous version in fact also requires an instance, though. I moved the field to instance attributes to access it directly when default value are checked against the json schema.